api/v1/controllers/samples.js

```diff
@@ -143,11 +144,14 @@ module.exports = {
    * @param {Function} next - The next middleware function in the stack
    */
   upsertSample(req, res, next) {
-    utils.noReadOnlyFieldsInReq(req, helper);
+
+    // make the name post-able
+    const readOnlyFields = helper.readOnlyFields.filter((field) => field !== 'name');
```

Comment:
Can we just say something like delete helper.readOnlyFields.name and pass the rest?